reconcile/skupper_network/site_controller.py

```diff
@@ -0,0 +1,179 @@
+import copy
+from collections.abc import Mapping
+from typing import Any
+
+from reconcile.skupper_network.models import SkupperSite
+
+LABELS = {"app": "skupper-site-controller", "managed-by": "qontract-reconcile"}
+CONNECTION_TOKEN_LABELS = {"skupper.io/type": "connection-token"}
+CONFIG_NAME = "skupper-site"
+
+
+def site_config(site: SkupperSite) -> dict[str, Any]:
+    """Skupper site configmap."""
+    return dict(
+        apiVersion="v1",
+        kind="ConfigMap",
+        metadata=dict(
+            name=CONFIG_NAME,
+            labels=LABELS,
+        ),
+        data=site.config.as_configmap_data(),
+    )
+
+
+def site_token(name: str, labels: Mapping[str, str]) -> dict[str, Any]:
+    """Skupper site token secret."""
+    _labels = copy.deepcopy(LABELS)
+    _labels.update(labels)
+    _labels["skupper.io/type"] = "connection-token-request"
+    return dict(
+        apiVersion="v1",
+        kind="Secret",
+        metadata=dict(
+            name=name,
+            labels=_labels,
+        ),
+    )
+
+
+def site_controller_deployment(site: SkupperSite) -> dict[str, Any]:
+    """Return skupper site controller deployment."""
+    return dict(
+        apiVersion="apps/v1",
+        kind="Deployment",
+        metadata=dict(
+            name="skupper-site-controller",
+            annotations={
+                "kube-linter.io/ignore-all": "ignore",
+            },
+            labels=LABELS,
+        ),
+        spec=dict(
+            replicas=1,
+            selector=dict(
+                matchLabels={"application": "skupper-site-controller"},
+            ),
+            template=dict(
+                metadata=dict(
+                    labels={"application": "skupper-site-controller"},
+                ),
+                spec=dict(
+                    serviceAccountName="skupper-site-controller",
+                    containers=[
+                        dict(
+                            name="site-controller",
+                            image=site.skupper_site_controller,
+                            env=[
+                                dict(
+                                    name="WATCH_NAMESPACE",
+                                    valueFrom=dict(
+                                        fieldRef=dict(
+                                            fieldPath="metadata.namespace",
+                                        ),
+                                    ),
+                                ),
+                            ],
+                        ),
+                    ],
+                ),
+            ),
+        ),
+    )
+
+
+def site_controller_service_account(
+    site: SkupperSite,
```

Comment:
do you need the `site` parameter here?

The same applies to a few methods in this module...

Reply:
You're right; it isn't needed.